bool enable_small_broadcast
bool enable_medium_broadcast
bool enable_large_broadcast